### PR TITLE
feat: instance attach/detach volume

### PR DESCRIPTION
**Pull request by DenBeke**

### CODE REVIEW

File: api/instance/v1/instance_utils.go [hunk 1]
```diff
@@ -86,3 +88,100 @@ func (s *API) UpdateIP(req *UpdateIPRequest, opts ...scw.RequestOption) (*Update
...
 }
+
+// AttachVolumeRequest contains the parameters to attach a volume to a server
+type AttachVolumeRequest struct {
+	Zone     utils.Zone `json:"-"`
+	ServerID string     `json:"-"`
+	VolumeID string     `json:"-"`
+}
+
+// AttachVolumeResponse [TODO]
```
Comment (kindermoumoute) [inline api/instance/v1/instance_utils.go]: TODO

File: api/instance/v1/instance_utils.go [hunk 2]
```diff
@@ -86,3 +88,100 @@ func (s *API) UpdateIP(req *UpdateIPRequest, opts ...scw.RequestOption) (*Update
...
+	})
+	if err != nil {
+		return nil, err
+	}
+	volumes := getServerResponse.Server.Volumes
+
+	newVolumes := volumesToVolumeTemplates(volumes)
+
+	// add volume to volumes list
+	newVolumes[fmt.Sprintf("%d", len(volumes))] = &VolumeTemplate{ID: req.VolumeID, Name: req.VolumeID /* name is ignored on this PATCH */}
```
Comment (QuentinBrosse) [inline api/instance/v1/instance_utils.go]:
```golang
key := fmt.Sprintf("%d", len(volumes))
newVolumes[key] = &VolumeTemplate{
    ID: req.VolumeID,
    // name is ignored on this PATCH
    Name: req.VolumeID
}
```

File: api/instance/v1/instance_utils.go [hunk 3]
```diff
@@ -86,3 +88,100 @@ func (s *API) UpdateIP(req *UpdateIPRequest, opts ...scw.RequestOption) (*Update
...
+		return nil, err
+	}
+
+	return &AttachVolumeResponse{Server: updateServerResponse.Server}, nil
+}
+
+// DetachVolumeRequest contains the parameters to detach a volume from a server
+type DetachVolumeRequest struct {
+	Zone     utils.Zone `json:"-"`
+	ServerID string     `json:"-"`
```
Comment (jerome-quere) [inline api/instance/v1/instance_utils.go]: Do we need server id ? Can this be get with /volume/id ?

File: api/instance/v1/instance_utils_test.go [hunk 4]
```diff
@@ -28,93 +28,127 @@ func TestAttachIP(t *testing.T) {
 		reverse      = "1.1.1.1"
 	)
 
-	// Create server
-	createServerResponse, err := instanceAPI.CreateServer(&CreateServerRequest{
-		Zone:         zone,
-		Name:         "instance_utils_test",
-		Organization: organization,
-		Image:        image,
+	t.Run("create server", func(t *testing.T) {
```
Comment (jerome-quere) [inline api/instance/v1/instance_utils_test.go]: why is this test in `TestAttachIP` ?

Comment (DenBeke) [reply api/instance/v1/instance_utils_test.go]: I reused added it to the test scenario (to reuse setup/teardown). I should be change the name of the function.

File: api/instance/v1/instance_utils.go (hunk 3 above)
Comment (DenBeke) [reply api/instance/v1/instance_utils.go]: If we do one more call we could omit the `ServerID`.

Comment (jerome-quere) [reply api/instance/v1/instance_utils.go]: Let's do one more call

Comment (DenBeke) [reply api/instance/v1/instance_utils.go]: 👍

File: api/instance/v1/instance_utils.go [hunk 5]
```diff
@@ -86,3 +88,117 @@ func (s *API) UpdateIP(req *UpdateIPRequest, opts ...scw.RequestOption) (*Update
...
+	// get volume
+	getVolumeResponse, err := s.GetVolume(&GetVolumeRequest{
+		Zone:     req.Zone,
+		VolumeID: req.VolumeID,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if getVolumeResponse.Volume == nil || getVolumeResponse.Volume.Server == nil {
+		return nil, fmt.Errorf("expected volume to have value in response")
```
Comment (jerome-quere) [inline api/instance/v1/instance_utils.go]: we should slip in to different error.
// expected volume to have value in response

and 
// server should be attach to a server

Comment (jerome-quere) [reply api/instance/v1/instance_utils.go]: if if getVolumeResponse.Volume == nil {
return nil, fmt.Errorf("expected volume to have value in response")
}

if (getVolumeResponse.Volume.Server == nil) {
return nil, fmt.Errorf("server should be attach to a server")
}

**Comment by jerome-quere** [comment]
fix #34

**Pull request merged**